src/DataDictionary/search/DictionarySearcher/DictionarySearcher.jsx

```diff
@@ -0,0 +1,205 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import AutoComplete from '@gen3/ui-component/dist/components/AutoComplete';
+import { prepareSearchData, searchKeyword, getSearchSummary } from './searchHelper';
+import './DictionarySearcher.css';
+
+class DictionarySearcher extends React.Component {
+  constructor(props) {
+    super(props);
+    this.searchData = prepareSearchData(props.dictionary);
+    this.autoCompleteRef = React.createRef();
+    this.state = {
+      suggestionList: [],
+      isSearchFinished: false,
+      searchResult: {
+        matchedNodes: [],
+        summary: {},
+      },
+      searchError: '',
+      hasError: false,
+    };
+    if (this.props.currentSearchKeyword) {
+      this.search(this.props.currentSearchKeyword);
+    }
+  }
+
+  componentDidMount() {
+    // resume search status after switching back from other pages
+    if (this.props.currentSearchKeyword) {
+      this.autoCompleteRef.current.setInputText(this.props.currentSearchKeyword);
+      this.search(this.props.currentSearchKeyword);
+      this.forceUpdate();
```

Comment:
do you need a forced update here? `setState` is being called in `this.search` so it should re-render?